### PR TITLE
Add wifi-router example

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

**Comment by scaprile** [comment]
Title is misleading, I was expecting to find some routing functionality... we should add "dashboard" to the title.
More comments when I build and see this

**Comment by scaprile** [comment]
Can we keep the password structure the same as the device dashboard ? admin, pass0

**Comment by scaprile** [comment]
And this is related to #2204 
```
./example
26027  1 mongoose.c:4570:mg_open_listen setsockopt(SO_REUSEADDR): 10022
26027  1 mongoose.c:3601:mg_listen      Failed: http://0.0.0.0:8000, errno 0
```

File: examples/wifi-router/Makefile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  DELETE = cmd /C del /Q /F /S  # Command prompt command to delete files
+  OUT ?= -o $(PROG)             # Build output
+  CFLAGS += -DSIGPIPE=0         # No SIGPIPE on Windows
+endif
+
+all: $(PROG)              # Default target. Build and run program
+	$(RUN) ./$(PROG) $(ARGS)
+
+web_root/bundle.js:
+	curl -s https://npm.reversehttp.com/preact,preact/hooks,htm/preact,preact-router -o $@
```
[the remark this answers is not in the document]
Comment (cpq) [reply examples/wifi-router/Makefile:23]: I think it should !

File: examples/wifi-router/Makefile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+# npm -g i tailwindcss tailwindcss-font-inter
+web_root/main.css: web_root/index.html web_root/main.js
+	npx tailwindcss -o $@ --minify
```
[the remark this answers is not in the document]
Comment (cpq) [reply examples/wifi-router/Makefile:27]: Yes. I have removed everything from the README and keep it blank.

**Review by scaprile — COMMENTED** [review]
more comments

File: examples/wifi-router/Makefile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+packed_fs.c: $(shell find web_root -type f) Makefile web_root/main.css web_root/bundle.js
+	$(CC) ../../test/pack.c -o pack
+	./pack `find web_root -type f` > $@
```
[the remark this answers is not in the document]
Comment (cpq) [reply examples/wifi-router/Makefile:31]: Made it work on Windows and UNIX
So far, we're doing it wrong, if you're referring to the device-dashboard

File: examples/wifi-router/README.md
[the remark this answers is not in the document]
Comment (cpq) [reply examples/wifi-router/README.md:1]: Cleared README.

File: examples/wifi-router/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+![](screenshots/login.webp)
+
+## Main dashboard
+
+The main dashboard page shows the interactive MQTT console
+
+![](screenshots/dashboard.webp)
+
+
+See a detailed tutorial at https://mongoose.ws/tutorials/device-dashboard/
```
[the remark this answers is not in the document]
Comment (cpq) [reply examples/wifi-router/README.md:29]: Cleared

File: examples/wifi-router/main.c [hunk 5]
```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2020-2022 Cesanta Software Limited
+// All rights reserved
+
+#include "web.h"
```
[the remark this answers is not in the document]
Comment (cpq) [reply examples/wifi-router/main.c:4]: Yes! web.c and web.h

**Review by scaprile — APPROVED** [review]

**Review by scaprile — CHANGES_REQUESTED** [review]
- The name is misleading, I was expecting to find some routing functionality... we should add "dashboard" to it.
- Can we keep the password structure the same as the device dashboard ? admin, pass0
- unit test fails

File: mongoose.c [hunk 6]
```diff
@@ -4554,8 +4554,7 @@ bool mg_open_listener(struct mg_connection *c, const char *url) {
                                 (char *) &on, sizeof(on))) != 0) {
       // "Using SO_REUSEADDR and SO_EXCLUSIVEADDRUSE"
       MG_ERROR(("setsockopt(SO_EXCLUSIVEADDRUSE): %d %d", on, MG_SOCK_ERR(rc)));
-#endif
-#if defined(SO_REUSEADDR) && (!defined(LWIP_SOCKET) || SO_REUSE)
+#elif defined(SO_REUSEADDR) && (!defined(LWIP_SOCKET) || SO_REUSE)
     } else if ((rc = setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, (char *) &on,
```
Comment (scaprile) [inline mongoose.c:4558]: This works, but unit test now fails.

**Pull request merged**